forum/users/permissions.py

```diff
@@ -0,0 +1,29 @@
+from rest_framework.permissions import BasePermission
+from rest_framework.request import Request
+from .models import Role, CustomUser
+
+class HasRole(BasePermission):
+    """
+    Permission class to check if a user has the required role,
+    based on both the token and the database state.
+    """
+
+    def __init__(self, role: Role):
+        self.role = role
```

Comment:
Permission classes should generally be stateless. Defining `__init__` with a dynamic parameter like `role` makes the permission class less reusable and deviates from standard DRF patterns.
DRF permissions are instantiated once and used across multiple requests. Using `__init__` dynamically can lead to unintended side effects.